hazelcast/src/main/java/com/hazelcast/core/RingbufferStore.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.core;
+
+import java.util.Map;
+
+/**
+ * Ringbuffer store makes a ring buffer backed by a central data store; such as database, disk, etc.
+ *
+ * @param <T> ring buffer item type
+ */
+public interface RingbufferStore<T> {
+
+    /**
+     * Stores one item with it's corresponding sequence.
+     *
+     * @param sequence the sequence ID of the data
+     * @param data     the value of the data to store
+     */
+    void store(long sequence, T data);
+
+    /**
+     * Stores multiple entries. Implementation of this method can optimize the store operation.
+     *
+     * @param map map of entries to store
+     */
+    void storeAll(Map<Long, T> map);
+
+    /**
+     * Loads the value of a given sequence. Null value means that the item was not found.
+     *
+     * @param sequence the sequence of the requested item
+     * @return requested item, null if not found
+     */
+    Object load(long sequence);
```

Comment:
Not yet. I am thinking of a use case - perhaps when a user requests a sequence and that sequence is before the head sequence, the container could load up to bulk_size items from the store and keep them in memory so that the subsequent loads are faster. The problem is, though, that other subscribers could be too slow and they could request some other sequence ID, thus loading some other sequence range. If we have two slow subscribers, we could end up by alternating loading ranges of items. Or keep them both in memory. But then I would suggest dropping them as they are read so we don't keep them indefinitely. And that has issues too if we have two slow subscribers. 
The queue store doesn't have this problem because it bulk loads for the ID's in the queue and no other ID's.
So I believe the bulk load could be beneficial but I'm not sure yet how to implement it correctly. Any suggestions? We could always add it later if necessary...